<?xml version="1.0"?>
<launch>

	<rosparam command="load" file="$(find retail_store_planning)/config/waypoints.yaml" />
	<rosparam command="load" file="$(find retail_store_planning)/config/objects.yaml" />
	<rosparam command="load" file="$(find retail_store_planning)/config/place.yaml" />

	<!-- ROSPlan -->
	<include file="$(find rosplan_planning_system)/launch/interfaced_planning_system.launch" >
		<arg name="domain_path"		value="$(find retail_store_planning)/pddl_files/domain_place2.pddl" />
		<arg name="problem_path"	value="$(find retail_store_planning)/pddl_files/problem_place2.pddl" />
	</include>

	
	<!-- move_base action interface -->
	<node name="rosplan_interface_move_base" pkg="retail_store_planning" type="moveBaseInterface" respawn="false" output="screen">
		<param name="knowledge_base"		value="rosplan_knowledge_base" />
		<param name="pddl_action_name"		value="move" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>

	<!-- pick action interface -->
	<node name="rosplan_interface_pick" pkg="retail_store_planning" type="pickInterface" respawn="false" output="screen">
		<param name="knowledge_base"		value="rosplan_knowledge_base" />
		<param name="pddl_action_name"		value="pick" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>

	<!-- place action interface -->
	<node name="rosplan_interface_place" pkg="retail_store_planning" type="placeInterface" respawn="false" output="screen">
		<param name="knowledge_base"		value="rosplan_knowledge_base" />
		<param name="pddl_action_name"		value="place" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>

	<!-- place1 action interface -->
	<node name="rosplan_interface_place1" pkg="retail_store_planning" type="placeInterface" respawn="false" output="screen">
		<param name="knowledge_base"		value="rosplan_knowledge_base" />
		<param name="pddl_action_name"		value="place1" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>

	<!-- place2 action interface -->
	<node name="rosplan_interface_place2" pkg="retail_store_planning" type="placeInterface" respawn="false" output="screen">
		<param name="knowledge_base"		value="rosplan_knowledge_base" />
		<param name="pddl_action_name"		value="place2" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>

	<!-- Running the python file to generate the plan -->
	<node name="load_product_info" pkg="retail_store_planning" type="load_product_info.py" output="screen">
		<param name="cmd" value="$(find retail_store_planning)/load_product_info.py" />
	</node>
	

</launch>